academy/xl-run-prod/2-keys.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Prepare a validator and keys
+order: 2
+description: Set up keys for use for your validator
+tag: deep-dive
+---
+
+# Prepare a validator and keys
+
+In the [previous section](./1-software.md), you prepared a binary for your nodes. Some of your nodes will be validators. To propose and sign blocks, validators need access to private keys on an on-going basis. A regular node does not need such keys.
+
+Here you learn how to prepare a validator and handle its keys. This works whether you are preparing a validator to join a preexisting network, or you are setting up your validator to be part of the [genesis](./genesis.md).
+
+## Private key security considerations
+
+More precisely than needing access to private keys, validators only need the capability to sign blocks on an on-going basis. There is a security difference between access to the private key and access to a signing facility:
+
+1. When your validator has access to the private key, if your validator node has been compromised, your private key is too, and you are at the risk of wrongfully signing malicious blocks **for ever**.
+2. On the other hand, when you only provide a signing _service_ to your validator, and your validator node has been compromised, you are _only_ at the risk of wrongfully signing malicious blocks **for as long as the signing service is up**.
+
+In order to mitigate **point 1**, you can keep your private key in a [hardware security module](https://hub.cosmos.network/main/validators/validator-faq.html#how-to-handle-key-management) (a.k.a. HSM), from which it can be retrieved only once, during the HSM's offline setup. This HSM device then remains plugged into the computer that runs the validator or the signing service. See [here](https://hub.cosmos.network/main/validators/security.html#key-management-hsm) for the current list of supported devices. To use an HSM you own, you need physical access to the computer into which you plug it in.
```

Comment:
I'm not fully clear on how this mitigates point 1. Is it that the HSM can be physically unplugged during a crisis, so that the validator becomes broken and unable to sign?

Reply:
The HSM is plugged into the computer and the HSM has the private key on its own chip. The private key therefore cannot be read. Of course it can be used for as long as the computer is up and the HSM plugged in. But used is not the same as stolen.

Reply:
I see: the key itself is no longer open to theft, so it cannot be used independently of the physical component.